File: app/src/main/java/com/example/androidautobuildapk/MainActivity.java
package com.example.androidautobuildapk;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.view.Gravity;
import android.widget.*;
import android.graphics.Color;

public class MainActivity extends Activity {
    static { System.loadLibrary("vuzt_native"); }
    
    // Nama fungsi ini harus sama persis dengan yang di lib.rs (Robusta)
    private native String mesinPusatRust(String input);

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        LinearLayout root = new LinearLayout(this);
        root.setOrientation(LinearLayout.VERTICAL);
        root.setGravity(Gravity.CENTER);
        root.setPadding(64, 64, 64, 64);
        root.setBackgroundColor(Color.parseColor("#121212"));

        // Judul
        TextView title = new TextView(this);
        title.setText("VUZT RUST BRIDGE");
        title.setTextColor(Color.CYAN);
        title.setPadding(0, 0, 0, 40);
        root.addView(title);

        // Input Text
        final EditText inputField = new EditText(this);
        inputField.setHint("Ketik sesuatu untuk Rust...");
        inputField.setHintTextColor(Color.GRAY);
        inputField.setTextColor(Color.WHITE);
        root.addView(inputField);

        // Hasil dari Rust
        final TextView display = new TextView(this);
        display.setText("Menunggu hasil...");
        display.setTextColor(Color.LTGRAY);
        display.setPadding(0, 40, 0, 40);
        display.setGravity(Gravity.CENTER);
        root.addView(display);

        // Tombol Proses
        Button btn = new Button(this);
        btn.setText("KIRIM KE RUST");
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String text = inputField.getText().toString();
                if(!text.isEmpty()) {
                    // Panggil fungsi Rust
                    String hasil = mesinPusatRust(text);
                    display.setText(hasil);
                    display.setTextColor(Color.GREEN);
                }
            }
        });
        root.addView(btn);

        setContentView(root);
    }
}

File: app/src/main/java/com/example/androidautobuildapk/MainActivity.kt
package com.example.androidautobuildapk

import android.app.Activity
import android.os.Bundle
import android.widget.*
import android.view.Gravity
import android.graphics.Color
import android.graphics.Typeface

class MainActivity : Activity() {

    companion object {
        init { System.loadLibrary("vuzt_native") }
    }

    private external fun mesinPusatRust(perintah: String): String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val root = LinearLayout(this).apply {
            orientation = LinearLayout.VERTICAL
            gravity = Gravity.CENTER
            setPadding(40, 40, 40, 40)
            setBackgroundColor(Color.parseColor("#121212"))
        }

        // --- UI Komponen ---
        val logView = TextView(this).apply {
            text = "SYSTEM LOG: READY"
            textSize = 12f
            setTextColor(Color.GRAY)
            typeface = Typeface.MONOSPACE
            setPadding(0, 0, 0, 50)
        }

        val statusView = TextView(this).apply {
            text = "IDLE"
            textSize = 22f
            setTextColor(Color.WHITE)
            gravity = Gravity.CENTER
            setPadding(0, 0, 0, 80)
        }

        // --- Fungsi Helper untuk Eksekusi Pipa ---
        fun kirimInstruksi(cmd: String) {
            try {
                val rawRespon = mesinPusatRust(cmd)
                // Parsing sederhana hasil dari Rust (STATE:XX|COLOR:YY|MSG:ZZ)
                val parts = rawRespon.split("|")
                val state = parts[0].replace("STATE:", "")
                val color = parts[1].replace("COLOR:", "")
                val msg = parts[2].replace("MSG:", "")

                statusView.text = state
                statusView.setTextColor(Color.parseColor(color))
                logView.text = "LOG: $msg"
                
            } catch (e: Exception) {
                logView.text = "LOG ERROR: ${e.message}"
            }
        }

        // --- Tombol-Tombol ---
        val btnProses = Button(this).apply { text = "EKSEKUSI DATA" }
        val btnSetting = Button(this).apply { text = "PENGATURAN" }
        val btnHistory = Button(this).apply { text = "RIWAYAT" }

        btnProses.setOnClickListener { kirimInstruksi("PROSES") }
        btnSetting.setOnClickListener { kirimInstruksi("SETTING") }
        btnHistory.setOnClickListener { kirimInstruksi("HISTORY") }

        // Tambah ke Layout
        root.addView(logView)
        root.addView(statusView)
        root.addView(btnProses)
        root.addView(btnSetting)
        root.addView(btnHistory)

        setContentView(root)
    }
}
